execute/tokendata/token_data.go

```diff
@@ -0,0 +1,181 @@
+package tokendata
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"time"
+
+	"github.com/patrickmn/go-cache"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+
+	"github.com/smartcontractkit/chainlink-ccip/pkg/reader"
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+)
+
+const (
+	// cacheExpiration sets how long we keep the seen attestations in the cache for computing duration.
+	// We can't keep them forever, so we don't track messages which are waiting for longer than 70 minutes
+	cacheExpiration = 70 * time.Minute
+	cacheCleanup    = 10 * time.Minute
+)
+
+var (
+	ErrDataMissing     = errors.New("token data missing")
+	ErrNotReady        = errors.New("token data not ready")
+	ErrRateLimit       = errors.New("token data API is being rate limited")
+	ErrTimeout         = errors.New("token data API timed out")
+	ErrUnknownResponse = errors.New("unexpected response from attestation API")
+
+	promAttestationDurations = promauto.NewHistogramVec(
```

Comment:
Could you let me know why you moved them here? To be honest, I think the previous approach had better isolation. `observability` stuff was moved to a separate file and therefore not pollute common code. These `cacheExpirations` and `prom` objects add unnecessary cognitive load here